qiskit/pulse/schedule.py

```diff
@@ -274,6 +274,21 @@ def parameters(self) -> Set:
         """Parameters which determine the schedule behavior."""
         return self._parameter_manager.parameters
 
+    @property
+    def pulses(self):
```

Comment:
This shouldn't be a property because it internally runs some logic. Instead of having two different methods, you can combine them with the following pattern.

```python
def get_pulses(self, name: Optional[str] = None):
    if not name:
        # return all pulses
    # otherwise return specified pulses
```